diff -pubr sipp-3.7.1.orig/CMakeLists.txt sipp-3.7.1/CMakeLists.txt
--- sipp-3.7.1.orig/CMakeLists.txt	2023-05-19 21:21:53.000000000 +0100
+++ sipp-3.7.1/CMakeLists.txt	2023-07-24 11:10:20.812071916 +0100
@@ -24,6 +24,7 @@ option(USE_SSL "Build with SIPS support"
 option(USE_SCTP "Build with SCTP support" OFF)
 option(USE_PCAP "Build with PCAP playback support" OFF)
 option(USE_GSL "Build with improved statistical support" ON)
+option(USE_UUID "Build with UUID support" OFF)
 
 file(GLOB all_SRCS
   "${PROJECT_SOURCE_DIR}/src/*.cpp"
@@ -120,6 +121,10 @@ if(USE_SCTP)
   add_definitions("-DUSE_SCTP")
 endif(USE_SCTP)
 
+if(USE_UUID)
+  add_definitions("-DUSE_UUID")
+endif(USE_UUID)
+
 # add the executable
 link_directories("/usr/local/lib")
 add_executable(sipp ${all_SRCS} "${PROJECT_SOURCE_DIR}/src/sipp.cpp")
@@ -252,4 +257,14 @@ if(USE_SCTP)
   endif()
 endif()
 
+if(USE_UUID)
+  find_library(UUID_LIBRARY uuid)
+  if(UUID_LIBRARY)
+    target_link_libraries(sipp ${UUID_LIBRARY})
+    target_link_libraries(sipp_unittest ${UUID_LIBRARY})
+  else()
+    message(FATAL_ERROR "libuuid was not found; pleast install devel package")
+  endif()
+endif()
+
 install(TARGETS sipp DESTINATION bin)
diff -pubr sipp-3.7.1.orig/build.sh sipp-3.7.1/build.sh
--- sipp-3.7.1.orig/build.sh	2023-05-19 21:21:53.000000000 +0100
+++ sipp-3.7.1/build.sh	2023-07-24 11:09:12.492068706 +0100
@@ -19,9 +19,9 @@ fi
 if test "$*" = "--none"; then
     cmake . -DUSE_GSL=
 elif test "$*" = "--common"; then
-    cmake . -DUSE_GSL=1 -DUSE_PCAP=1 -DUSE_SSL= -DUSE_SCTP=
+    cmake . -DUSE_GSL=1 -DUSE_UUID=1 -DUSE_PCAP=1 -DUSE_SSL= -DUSE_SCTP=
 elif test "$*" = "--full"; then
-    cmake . -DUSE_GSL=1 -DUSE_PCAP=1 -DUSE_SSL=1 -DUSE_SCTP=1
+    cmake . -DUSE_GSL=1 -DUSE_UUID=1 -DUSE_PCAP=1 -DUSE_SSL=1 -DUSE_SCTP=1
 else
     # Debug build? Add -DDEBUG=1
     # Adjusted SIP max size? Add -DSIPP_MAX_MSG_SIZE=262144
Only in sipp-3.7.1: build.sh.orig
diff -pubr sipp-3.7.1.orig/src/call.cpp sipp-3.7.1/src/call.cpp
--- sipp-3.7.1.orig/src/call.cpp	2023-05-19 21:21:53.000000000 +0100
+++ sipp-3.7.1/src/call.cpp	2023-07-24 11:09:12.492068706 +0100
@@ -52,6 +52,10 @@
 #include <sys/wait.h>
 #include <string>
 
+#ifdef USE_UUID
+#include <uuid/uuid.h>
+#endif
+
 #ifdef PCAPPLAY
 #include "send_packets.h"
 #endif
@@ -804,6 +808,11 @@ call *call::add_call(int userId, bool ip
 {
     static char call_id[MAX_HEADER_LEN];
 
+#ifdef USE_UUID
+    uuid_t uuid;
+    char uuid_str[37];
+#endif
+
     const char * src = call_id_string;
     int count = 0;
 
@@ -815,6 +824,13 @@ call *call::add_call(int userId, bool ip
         if (*src == '%') {
             ++src;
             switch(*src++) {
+#ifdef USE_UUID
+            case 'U':
+                uuid_generate(uuid);
+                uuid_unparse(uuid, uuid_str);
+                count += snprintf(&call_id[count], MAX_HEADER_LEN-count-1,"%s", uuid_str);
+                break;
+#endif
             case 'u':
                 count += snprintf(&call_id[count], MAX_HEADER_LEN-count-1, "%u", next_number);
                 break;
Only in sipp-3.7.1/src: call.cpp.orig
diff -pubr sipp-3.7.1.orig/src/sipp.cpp sipp-3.7.1/src/sipp.cpp
--- sipp-3.7.1.orig/src/sipp.cpp	2023-05-19 21:21:53.000000000 +0100
+++ sipp-3.7.1/src/sipp.cpp	2023-07-24 11:09:12.502068707 +0100
@@ -229,7 +229,7 @@ struct sipp_option options_table[] = {
     {"", "Call behavior options:", SIPP_HELP_TEXT_HEADER, NULL, 0},
     {"aa", "Enable automatic 200 OK answer for INFO, NOTIFY, OPTIONS and UPDATE.", SIPP_OPTION_SETFLAG, &auto_answer, 1},
     {"base_cseq", "Start value of [cseq] for each call.", SIPP_OPTION_CSEQ, NULL, 1},
-    {"cid_str", "Call ID string (default %u-%p@%s).  %u=call_number, %s=ip_address, %p=process_number, %%=% (in any order).", SIPP_OPTION_STRING, &call_id_string, 1},
+    {"cid_str", "Call ID string (default %u-%p@%s).  %u=call_number, %U=uuid, %s=ip_address, %p=process_number, %%=% (in any order).", SIPP_OPTION_STRING, &call_id_string, 1},
     {"d", "Controls the length of calls. More precisely, this controls the duration of 'pause' instructions in the scenario, if they do not have a 'milliseconds' section. Default value is 0 and default unit is milliseconds.", SIPP_OPTION_TIME_MS, &duration, 1},
     {"deadcall_wait", "How long the Call-ID and final status of calls should be kept to improve message and error logs (default unit is ms).", SIPP_OPTION_TIME_MS, &deadcall_wait, 1},
     {"auth_uri", "Force the value of the URI for authentication.\n"
Only in sipp-3.7.1/src: sipp.cpp.orig
